system/Router/Router.php

```diff
@@ -404,10 +404,13 @@ protected function checkRoutes(string $uri): bool
 		// Loop through the route array looking for wildcards
 		foreach ($routes as $key => $val)
 		{
+			$localeSegment = NULL; // remove stale value 
 			// Are we dealing with a locale?
 			if (strpos($key, '{locale}') !== false)
 			{
-				$localeSegment = array_search('{locale}', explode('/', $key));
+				$tempkey = preg_replace( '#\(([^()]*)\)#i' , 'x' , $key);
```

Comment:
temporary replacing regex with 'x' just to remove '/' character

Reply:
This might work for your usecase, but dpesn't feel right for the general usecase. I agree that your other two proposed solutions would be better. The second suggestion, handling locale separately form the rest of the URI, sounds most promising.

Would another workaround be to put the locale in the second segment if it is to be used?

Reply:
That is good to place locale there. But some user just remove locale segment (they rely on website's link to change locale instead on typing on the address bar) and website rely on default locale.

So my usecase is put locale segment on last segment

Reply:
Maybe the routing needs to handle the special case the the locale might be the last segment instead of the first one then? Actually, it needs to woek either way.
Hmm - this might provide the basis for a unit test to expose the problem :)
Either way, simply replacing {locale} with x doesn't seem right :-/

Reply:
Sure it not right :( , it is just quick hack. In the end, I just use GET variable to change locale and save it to visitor's cookie. My routing has assign 0 - 3 segment already, so locale segment can misinterpret as path segment. 
If locale always shown on 4th segment, that will be easier todo. Unfortunately, path with 2 segment can also have {locale} and misinterpret as 3rd segment
This is the cost to make url looks memorable :)

In the end, I just remove routing with {locale} and change locale using GET instead and save it to visitor's cookie

Reply:
Anyway, if {locale} segment gets overlap with path / parameter segment, that is developer problems, nothing todo with framework. Naturally it will get misinterpret, which comes first. 
{locale} may misinterpret as another regular segment.
or ordinary segment misinterpret as locale code (surely will fallback to default locale, but the regular segment won't get executed)